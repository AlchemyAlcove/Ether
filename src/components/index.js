import Button from "./button";
import Ellipsis from "./ellipsis";
import Notice from "./notice";

export{
  Button,
  Ellipsis,
  Notice
};
