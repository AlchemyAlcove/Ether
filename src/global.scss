$containerWidth: 1200px;
$desktopBreakpoint: 992px;
$headerHeight: 50px;
$tabletBreakpoint: 768px;

$background: #ECF0F1;
$headerBackground: #363D44;
$headerBackgroundHighlight: #24292D;
$headerDropdownBackground: #FFFFFF;
$headerLogoHighlight: #1EB6F2;
$headerText: #FFFFFF;
$headerTextOpaque: #BFC4CA;
$text: #333333;

$font: 'Open Sans', sans-serif;
$weightThin: 300;
$weightRegular: 400;
$weightThick: 700;

body, html {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  background: $background;
  font-size: 14px;
  font-family: $font;
  font-weight: $weightRegular;
  color: $text;

  * {
    box-sizing: border-box;
  }
}

*:focus {
  outline: 0;
}

a {
  color: $text;
  text-decoration: none;
  cursor: pointer;
}

@import "./components/ellipsis/ellipsis.scss";