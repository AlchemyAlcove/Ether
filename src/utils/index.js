import setVariables from "./setVariables";

export{
  setVariables
};
