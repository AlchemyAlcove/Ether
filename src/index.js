import { Button, Ellipsis, Notice } from "./components";
import { setVariables } from "./utils";
import { cx, isNil } from "@alchemyalcove/basic-functions";

export{
  cx,
  Button,
  Ellipsis,
  Notice,
  isNil,
  setVariables
};
