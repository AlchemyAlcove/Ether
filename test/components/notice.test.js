import { Notice } from "../../src";
import { fireEvent, render } from "react-testing-library";
import React from "react"; // eslint-disable-line no-unused-vars
import sinon from "sinon";

test("generic notice", () => {
  const{container} = render(<Notice/>);

  expect(container.firstChild).toMatchSnapshot();
});

test("secondary notice", () => {
  const{container} = render(<Notice color="secondary"/>);

  expect(container.firstChild).toMatchSnapshot();
});

test("success small outline button", () => {
  const{container} = render(<Notice color="success"/>);

  expect(container.firstChild).toMatchSnapshot();
});

test("danger small link button", () => {
  const{container} = render(<Notice color="danger"/>);

  expect(container.firstChild).toMatchSnapshot();
});

test("warning small button", () => {
  const{container} = render(<Notice color="warning"/>);

  expect(container.firstChild).toMatchSnapshot();
});
