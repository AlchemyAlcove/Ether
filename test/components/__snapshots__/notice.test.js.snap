// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`danger small link button 1`] = `
<div
  class="notice color-danger"
/>
`;

exports[`generic notice 1`] = `
<div
  class="notice color-primary"
/>
`;

exports[`secondary notice 1`] = `
<div
  class="notice color-secondary"
/>
`;

exports[`success small outline button 1`] = `
<div
  class="notice color-success"
/>
`;

exports[`warning small button 1`] = `
<div
  class="notice color-warning"
/>
`;
